modyn/supervisor/internal/pipeline_executor/evaluation_executor.py

```diff
@@ -292,40 +295,31 @@ def _single_evaluation(self, log: StageLog, eval_status_queue: Queue, eval_req:
                     self.grpc.init_evaluator()
                     raise e
 
+        def get_failure_reason(eval_aborted_reason: EvaluationAbortedReason) -> str:
+            return EvaluationAbortedReason.DESCRIPTOR.values_by_number[eval_aborted_reason].name
+
         if not response.evaluation_started:
-            log.info.failure_reason = EvaluationAbortedReason.DESCRIPTOR.values_by_number[
-                response.eval_aborted_reason
-            ].name
+            failure_reason = get_failure_reason(response.interval_responses[0].eval_aborted_reason)
             logger.error(
-                f"Evaluation for model {eval_req.id_model} on split {eval_req.interval_start} to "
-                f"{eval_req.interval_end} not started with reason: {log.info.failure_reason}."
+                f"Evaluation for model {model_id_to_eval} on split {interval_start} to "
+                f"{interval_end} not started with reason: {failure_reason}."
             )
-            return
+            return failure_reason, {}
 
-        logger.info(
-            f"Evaluation started for model {eval_req.id_model} on split {eval_req.interval_start} "
-            f"to {eval_req.interval_end}."
-        )
-        reporter = EvaluationStatusReporter(
-            eval_status_queue,
-            response.evaluation_id,
-            eval_req.dataset_id,
-            response.dataset_size,
-        )
-        evaluation = {response.evaluation_id: reporter}
-        reporter.create_tracker()
-        self.grpc.wait_for_evaluation_completion(eval_req.training_id, evaluation)
-
-        eval_result_writer = JsonResultWriter(self.pipeline_id, eval_req.trigger_id, self.pipeline_logdir)
-        self.grpc.store_evaluation_results([eval_result_writer], evaluation)
-        self.grpc.cleanup_evaluations([int(i) for i in evaluation])
-        assert isinstance(eval_result_writer, JsonResultWriter)
-
-        log.info.results = (
-            eval_result_writer.results["datasets"][0][dataset_config.dataset_id]
-            if eval_result_writer.results["datasets"]
-            else {}
-        )
+        logger.info(f"Evaluation started for model {model_id_to_eval} on split {interval_start} " f"to {interval_end}.")
+        self.grpc.wait_for_evaluation_completion(response.evaluation_id)
+        eval_data = self.grpc.get_evaluation_results(response.evaluation_id)
+        self.grpc.cleanup_evaluations([response.evaluation_id])
+        # here we assume only one interval is evaluated
+        eval_results: dict = {"dataset_size": response.interval_responses[0].dataset_size, "metrics": []}
+
+        for single_eval_data in eval_data:
+            if single_eval_data.interval_index == 0:
+                for metric in single_eval_data.evaluation_data:
+                    eval_results["metrics"].append({"name": metric.metric, "result": metric.result})
```

Comment:
I don't understand the if == 0 yet. Why is that valid?

Reply:
I realized it's probably due to the fact we only have a single interval, but then I am not sure why we have an if instead of just asserting it's 0 since otherwise we have a problem in the current implementation :D

Reply:
you are right. I was just being lazy in the sense that the result collection logic needs to be rewritten by @robinholzi anyway. But I will change it as you suggest!

Reply:
done